### PR TITLE
Fix: bug of training force and stress with DeePKS

### PR DESCRIPTION
**Pull request by dyzheng**
When user training DeePKS model with ABACUS, force and stress labels contain two gradient files named "grad_vx.npy" and "grad_vepsl.npy".
These two files are needed by DeePKS-kit but don't affect the self-consistent computation with loading DeePKS model in ABACUS.
Before https://github.com/deepmodeling/abacus-develop/pull/3444 , when "cal_stress = 1", the line of " https://github.com/deepmodeling/abacus-develop/blob/v3.5.0/source/module_hamilt_lcao/module_deepks/LCAO_deepks_pdm.cpp#L869 " conflict with the line of "https://github.com/deepmodeling/abacus-develop/blob/v3.5.0/source/module_hamilt_lcao/module_deepks/LCAO_deepks_pdm.cpp#L808" , which will cause severe bug that "grad_vx.npy" and "grad_vepsl.npy" are wrong and convergence of DeePKS model training won't be achieved.

This bug will be fixed after https://github.com/deepmodeling/abacus-develop/pull/3444 , I also add checklists in deepks test cases for protecting these code.

### CODE REVIEW

File: source/module_hamilt_lcao/module_deepks/LCAO_deepks_pdm.cpp [hunk 1]
```diff
@@ -805,9 +805,9 @@ void LCAO_Deepks::cal_gdmx_k(const std::vector<std::vector<std::complex<double>>
                     key_tuple key_2(ibt2,dR2.x,dR2.y,dR2.z);
 					for (int iw1l = 0; iw1l < row_indexes.size(); ++iw1l)
                     {
-                        std::vector<double> nlm1 = this->nlm_save_k[iat][key_1][row_indexes[iw1l]][0];
                         for (int iw2l = 0; iw2l < col_indexes.size(); ++iw2l)
                         {
+                            std::vector<double> nlm1 = this->nlm_save_k[iat][key_1][row_indexes[iw1l]][0];
```
Comment (WHUweiqingzhou) [inline source/module_hamilt_lcao/module_deepks/LCAO_deepks_pdm.cpp:810]: Plz add a UnitTest to trace similar error.

Comment (dyzheng) [reply source/module_hamilt_lcao/module_deepks/LCAO_deepks_pdm.cpp:810]: I have added check lists in deepks test cases for both multi-k and gamma-only cases.
The UnitTest will be implemented in future PRs.

**Review by WHUweiqingzhou — APPROVED** [review]
Good Debug!

**Comment by wenfei-li** [comment]
I am still not convinced by this pr, will raise an issue about this

**Pull request merged**